### PR TITLE
Use external JSON configuration files for PAYG

### PR DESCRIPTION
**Pull request by mackdk**
## What does this PR change?

This PR changes the PAYG logic to use the external JSON files generated by the [external timer task](https://github.com/uyuni-project/uyuni/pull/8691) to check compliance and populate PAYG data.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [X] **DONE**

## Test coverage
- Already covered. Unit tests were updated

- [X] **DONE**

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/24141 https://github.com/SUSE/spacewalk/issues/24140

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Review by cbbayburt — APPROVED** [review]
Looks good for Naica :+1:

File: java/code/src/com/suse/cloud/PaygComplainceInfo.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    private boolean paygInstance;
+
+    private boolean compliant;
+
+    private CloudProvider cloudProvider;
+
+    @SerializedName("hasModifiedPackages")
+    private boolean anyPackageModified;
+
+    private boolean billingServiceRunning;
```
Comment (mcalmer) [inline java/code/src/com/suse/cloud/PaygComplainceInfo.java:41]: I added now also "billingServiceStatus" to the json. This should be added here as well.
It is also a boolean value.

File: java/code/src/com/suse/cloud/CloudPaygManager.java [hunk 2]
```diff
@@ -155,226 +133,77 @@ public boolean checkRefreshCache(boolean force) {
...
 
+        // If the external compliance tool report it's not ok, it cannot be compliant
+        if (!complainceInfo.isCompliant()) {
+            LOG.error("The instance has been reported as non-compliant.");
+            if (complainceInfo.isAnyPackageModified()) {
+                LOG.error("Some packages have been modified and cannot be trusted.");
+            }
+            if (complainceInfo.isBillingServiceRunning()) {
+                LOG.error("The CSP Billing Adapter service is not active.");
+            }
```
Comment (mcalmer) [inline java/code/src/com/suse/cloud/CloudPaygManager.java:158]: The new BillingAdapterStatus should be used here as well.

File: java/code/src/com/suse/cloud/CloudPaygManager.java [hunk 3]
```diff
@@ -155,226 +133,77 @@ public boolean checkRefreshCache(boolean force) {
...
     private boolean detectHasSCCCredentials() {
         return CredentialsFactory.listSCCCredentials().stream()
                 .anyMatch(c -> mgr.isSCCCredentials(c));
     }
 
     private boolean detectIsCompliant() {
-        // do not use isPaygInstance() to prevent a loop
-        if (BooleanUtils.isFalse(isPaygInstance)) {
+        // Parse the compliance info from the external json file, if available
+        complainceInfo = getInstanceComplianceInfo();
```
Comment (mcalmer) [inline java/code/src/com/suse/cloud/CloudPaygManager.java:143]: We should add here also a check if the compliance file is valid.
It has also a timestamp and is refreshed also every 10 minutes like the payg.json

**Review by mcalmer — CHANGES_REQUESTED** [review]
Some changes needed due to last minute changes of the timer job

File: java/code/src/com/suse/cloud/PaygComplainceInfo.java (hunk 1 above)
Comment (mackdk) [reply java/code/src/com/suse/cloud/PaygComplainceInfo.java]: Done!

File: java/code/src/com/suse/cloud/CloudPaygManager.java (hunk 2 above)
Comment (mackdk) [reply java/code/src/com/suse/cloud/CloudPaygManager.java:167]: Done and also fixed the check (it should have been negated).

File: java/code/src/com/suse/cloud/CloudPaygManager.java (hunk 3 above)
Comment (mackdk) [reply java/code/src/com/suse/cloud/CloudPaygManager.java:145]: Added the validity check of before checking if it's BYOS.

**Review by mcalmer — APPROVED** [review]
LGTM - thanks

**Review by lucidd — APPROVED** [review]

**Pull request merged**